Tests/MongoSwiftSyncTests/ReadPreferenceOperationTests.swift

```diff
@@ -0,0 +1,46 @@
+@testable import MongoSwift
+import Nimble
+import TestsCommon
+import XCTest
+
+final class ReadPreferenceOperationTests: MongoSwiftTestCase {
+    override func setUp() {
+        self.continueAfterFailure = false
+    }
+
+    func testOperationReadPreference() throws {
+        // setup a collection
+        let client = try MongoClient.makeTestClient()
+        let db = client.db(type(of: self).testDatabase)
+        defer { try? db.drop() }
+        let coll = try db.createCollection(self.getCollectionName(suffix: "1"))
+
+        let command: Document = ["count": .string(coll.name)]
+
+        // expect runCommand to return a success response when passing in a valid read preference
+        let opts = RunCommandOptions(readPreference: ReadPreference(.secondaryPreferred))
+        let res = try db.runCommand(command, options: opts)
+        expect(res["ok"]?.asDouble()).to(equal(1.0))
+
+        // expect running other commands to not throw errors when passing in a valid read preference
+        expect(try coll.find(options: FindOptions(readPreference: ReadPreference(.primary)))).toNot(throwError())
+        expect(try coll.findOne(options: FindOneOptions(readPreference: ReadPreference(.primary)))).toNot(throwError())
+
+        expect(try coll.aggregate(
+            [["$project": ["a": 1]]],
+            options: AggregateOptions(readPreference: ReadPreference(.secondaryPreferred))
+        ))
+            .toNot(throwError())
```

Comment:
these `toNot`s seem misformatted, does swiftformat not move them? I feel like they look better like the second one.

Reply:
I agree, it looks weird. swiftformat doesn't move them but I've put them on the same line as the )).